sam/methods.go

```diff
@@ -15,7 +15,7 @@ func (s Sam) GetChromStart() int {
 
 func (s Sam) GetChromEnd() int {
 	var runLength int = 0
-	if s.Cigar[0].Op == '*' {
+	if s.Cigar == nil || s.Cigar[0].Op == '*' {
```

Comment:
Could this also be addressed by standardizing the value of s.Cigar when the read is unmapped?  Maybe it should always be a star, or always be nil (less memory allocation)?

Reply:
I agree that would be ideal, though I don't have the bandwidth for it at the moment. If someone else can handle it, then I will close this PR.